### ISSUE TITLE
How does model.save() differ from the following custom save function regarding optimizer params?

### ISSUE THREAD

**Issue by and-rewsmith** [keras-team/keras#9915]
Correct me if I'm wrong here, but I assume when model.fit() is called the optimizer's parameters (the learning rate, decay...) are restarted.

I want to avoid this so in my research I came across the following functions:

```
from keras.models import model_from_json

def save_model(model):    
    json_string = model.to_json()
    open('model.json', 'w').write(json_string)
    model.save_weights('weights.h5', overwrite=True)
    
def load_model():
    model = model_from_json(open('model.json').read())
    model.load_weights('weights.h5')
    sgd = SGD(lr = 0.1, momentum = 0.9, decay = 0, nesterov = False)
    model.compile(optimizer=sgd, loss = 'binary_crossentropy')
    return model
```

__Does saving the model with these functions restart the optimizer params? I am assuming it will however this code was posted as a solution so I'm confused.__
